lib/eslint.js

```diff
@@ -908,6 +908,8 @@ module.exports = (function() {
                 }
             }
 
+            ast.hasShebang = !!shebang;
```

Comment:
After a bit of digging I'm realizing that including the shebang in the tokens and comments store might be intentional and actually a good thing. Despite not being attached to any nodes or included in the `comments` array of the AST, transforming it into a standard JS line comment and including it in the store allows us to use `sourceCode.getTokenOrCommentBefore()` in rules to write rules around the shebang.

Had some thoughts and will write them in a comment below.